sdk/storage/azure-storage-blob/V10_V12.md

```diff
@@ -0,0 +1,379 @@
+# Storage Blob Service SDK Migration Guide from 10.x/11.x to 12.x
+
+In this section, we list the main changes you need to be aware of when converting your Storage Blob SDK library from Version 10/Version 11 to Version 12.
+
+## Prerequisites
+Java Development Kit (JDK) with version 8 or above
+
+## Reactor and ReactorNetty
+We've updated our asynchronous framework from RxJava to Reactor.
+For more info related to Reactor, please refer to [Project Reactor](https://projectreactor.io/)
+For more info of the motivation behind this major change, please refer to [this guide](https://github.com/Azure/azure-storage-java/blob/master/V12%20Upgrade%20Story.md).
+
+## Converting Core Classes
+Our core asynchronous classes have been replaced, as well as new synchronous counterparts added.
+
+| Core V10/11 classes  | Equivalent V12 Class | NEW synchronous clients |
+|---:|---:|---:|
+| ServiceURL  |  BlobServiceAsyncClient  | BlobServiceClient  |
+| ContainerURL |  ContainerAsyncClient | ContainerClient |
+| BlobURL    | BlobAsyncClient  |   BlobClient |
+| BlockBlobURL  |  BlockBlobAsyncClient  |  BlockBlobClient  |
+| AppendBlobURL  |  AppendBlobAsyncClient | AppendBlobClient  |
+| PageBlobURL |  PageBlobAsyncClient | PageBlobClient  |
```

Comment:
I think we should call out TransferManager here because people may not know where to look for file transfer methods otherwise.